xbmc/video/VideoInfoScanner.cpp

```diff
@@ -121,36 +121,45 @@ namespace VIDEO
       // result in unexpected behaviour.
       m_bCanInterrupt = false;
 
-      bool bCancelled = false;
-      while (!bCancelled && !m_pathsToScan.empty())
+
+      if (m_handle)
+        m_handle->SetTitle("Listing media files ...");
+
+      // List all media files of all 'm_pathsToScan'
+      //   INFO: m_pathsToScan may contain path with different library content, so we store their properties into a dedicated struct.
+      std::vector<SourcePathContent> _pathsContent;
+      CFileItemList remoteDirectoryItems[m_pathsToScan.size()];
+      int nbRemoteDirectoryItems = 0;
+
+      for (const auto& rootPathToScan : m_pathsToScan)
       {
-        /*
-         * A copy of the directory path is used because the path supplied is
-         * immediately removed from the m_pathsToScan set in DoScan(). If the
-         * reference points to the entry in the set a null reference error
-         * occurs.
-         */
-        std::string directory = *m_pathsToScan.begin();
-        if (m_bStop)
+        // Load root path properties from scraper settings:
+        bool foundDirectly = false;
+        bool bSkip = false;
+        SScanSettings rootPathScraperScanSettings;
+        ScraperPtr info = m_database.GetScraperForPath(rootPathToScan, rootPathScraperScanSettings, foundDirectly);
+        CONTENT_TYPE contentType_AsDefinedInScraper = info ? info->Content() : CONTENT_NONE;
```

Comment:
As a general rule, if you assign a value only once during lifetime of a variable, make it `const` to make this fact clear to the reader of the code.